### ISSUE TITLE
reproducible server 500 error from redirect url after successful login using Chrome

### ISSUE THREAD

**Issue by brianmay** [tanguilp/plugoid#14]
Unfortunately, logs are very vague, which is very strange. Normally I would expect a Phoenix stack trace.

```
20:21:44.827 [info]  GET /openid_connect_redirect_uri    
20:21:44.828 [info]  Sent 500 in 63Âµs
```

This seems to be 100% reproducible on Chrome from Linux and Chrome from Android. I try to access my website, get the login, login, and then get redirected to a 500 error.

Happens if I delete Cookies too.

On Linux, Chrome version 95.0.4638.69 (Official Build) (64-bit)

Meanwhile Firefox just works with the same website.

I will come back to this when I get time, but am a bit stumped for now.

**Comment by brianmay** [comment]
https://github.com/tanguilp/plugoid/blob/master/lib/plugoid/redirect_uri.ex#L163, the error is `Elixir.Plugoid.Session.StateSession.CookieNotFoundError`.

Oh, I see, we are trying to set "SameSite=None" on the cookie, but not setting Secure. Which is something modern versions of Chrome do not like and completely blocks creating the cookie.

https://web.dev/samesite-cookies-explained/#changes-to-the-default-behavior-without-samesite

Setting secure is going to be a pain for local testing, but I should be able to work something out.

Will test adding Secure to cookie and post here if it works.

**Comment by brianmay** [comment]
Also cookie not found error probably should be a 4xx error not a 5xx error.

**Comment by brianmay** [comment]
Oh, seems Secure cookies work fine now on http://localhost connections.

This is my current config which seems to work fine on localhost:

```
config :robotica_face, RoboticaFaceWeb.Endpoint,
  http: [
    port: port,
    ip: {0, 0, 0, 0, 0, 0, 0, 0},
    protocol_options: [max_header_value_length: 8096]
  ],

[...]

config :plugoid,
  auth_cookie_store: Plug.Session.COOKIE,
  auth_cookie_store_opts: [
    signing_salt: System.get_env("SIGNING_SALT")
  ],
  state_cookie_opts: [
    extra: "SameSite=None Secure"
  ],
  state_cookie_store_opts: [
    signing_salt: System.get_env("SIGNING_SALT")
  ]
```

**Comment by tanguilp** [comment]
Thanks for the report, browser support moves fast. Will fix ASAP.

`localhost` is often an exception in new protocols, which is nice.

**Comment by tanguilp** [comment]
Also I think you could do 

```elixir
  state_cookie_opts: [
    :secure,
    extra: "SameSite=None"
  ],
```
as this is a default option. Could you check? Not sure about the syntax, could be `secure: true` instead.

**Comment by tanguilp** [comment]
Try `secure: true` first :)

**Comment by tanguilp** [comment]
Fixed in 0.5.0
